src/components/BookForm/BookForm.js

```diff
@@ -6,77 +6,187 @@ import Row from "react-bootstrap/Row";
         <Form.Label>Описание</Form.Label>
-        <Form.Control as="textarea" rows={2} />
+        <Form.Control
+          value={book.discription}
+          name="discription"
+          onChange={onChangeForm}
+          as="textarea"
+          rows={2}
+        />
       </Form.Group>
 
-      {["Фантастика", "Детектив", "Драма"].map((label) => (
+      {CHECKBOXES.map(({ id }) => (
         <Form.Check
+          value={book.genre}
+          name="genre"
+          key={id}
           className="m-3"
           inline
-          label={label}
+          label={id}
           type="checkbox"
-          id={`inline-${label}-3`}
+          id={`inline-${id}-3`}
+          onChange={() => onChangeCheckbox(id)}
+          checked={book.genre.indexOf(id) > -1}
         />
       ))}
+      <Button onClick={() => checkAll()}>Чекбокс</Button>
 
       <Row className="m-2">
         <Form.Group as={Col} controlId="formGridCost">
           <Form.Label>Стоимость</Form.Label>
-          <Form.Control type="number" />
+          <Form.Control
+            value={book.cost}
+            name="cost"
+            onChange={onChangeForm}
+            type="number"
+          />
         </Form.Group>
 
         <Form.Group as={Col} controlId="formGridLanguage">
-          <Form.Label htmlFor="disabledSelect">Язык</Form.Label>
-          <Form.Select id="disabledSelect">
+          <Form.Label>Язык</Form.Label>
+          <Form.Select
+            value={book.language}
+            name="language"
+            onChange={onChangeForm}
+            id="disabledSelect"
+          >
+            <option disabled></option>
             <option>Украинский</option>
             <option>Английский</option>
           </Form.Select>
         </Form.Group>
 
         <div className="m-2">Наличие</div>
         <div className="availability">
-          {["Есть", "Нету"].map((label) => (
-            <div
-              key={`inline-${label}`}
-              className="m-1
-          "
-            >
+          {["Есть", "Нету"].map((availability) => (
+            <div key={availability} className="m-1">
               <Form.Check
+                value={availability}
+                onChange={onChangeForm}
                 inline
-                name="group1"
-                label={label}
+                name="availability"
+                label={availability}
                 type="radio"
-                id={`inline-${label}-1`}
+                checked={book.availability === availability}
               />
             </div>
           ))}
         </div>
         <Form.Group controlId="formFile" className="mb-3">
-          <Form.Control type="file" size="sm" />
+          <Form.Control
+            name="file"
+            type="file"
+            onChange={onImageChange}
+            size="sm"
+          />
+          <img
+            className="image"
+            src={book.file ? book.file : "no_photo.jpg"}
```

Comment:
В целом тоже хороший вариант
Можешь оставить
Но я хотел от тебя немного другое

```suggestion
            {
             book.file ? (<img src="...." />) : <span>no photo</span>
            }
```